utils_cv/detection/model.py

```diff
@@ -194,15 +234,21 @@ def __init__(
         if im_size is None:
             im_size = 500
 
-        self.device = torch_device()
+        self.device = device
+        if self.device is None:
+            self.device = torch_device()
+
         self.model = model
         self.dataset = dataset
         self.im_size = im_size
 
+        if dataset and "labels" in dataset.__dict__:
+            self.labels = dataset.labels
```

Comment:
This seems weird. We already store self.dataset which contains self.dataset.labels, and hence should probably not have a second reference via self.labels

Reply:
We allow `self.dataset is None`, so `add_labels()` is added to provide labels.

Reply:
I see. In your notebook, it uses add_labels because not dataset was specified:
   detector = DetectionLearner(model=get_pretrained_maskrcnn(), device=torch.device(DEVICE))
   detector.add_labels(coco_labels()[1:])

If add_labels HAS to be called (which I believe is the case) if no dataset was specified, then I would add a "labels" variable to the class constructor. And in the constructor it can throw an error if no dataset was provided and no "labels" variable.

Reply:
Make sense.  @jiata What do you think?

Reply:
Changed.